### PR TITLE
[CSSolver] Add test-cases for ambiguity issues resolved by weighted ranking

### PR DESCRIPTION
**Pull request by xedin**
Collects all (I could find) of the test-cases fixed by weighted ranking of solutions.

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

File: test/Sema/fixed_ambiguities/rdar35625473.swift [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// RUN: %target-swift-frontend -emit-sil -verify %s | %FileCheck %s
+
+// CHECK: function_ref @_T0s10CollectionPsE9dropFirst11SubSequenceQzSiF : $@convention(method) <τ_0_0 where τ_0_0 : Collection> (Int, @in_guaranteed τ_0_0) -> @out τ_0_0.SubSequence
+_ = [1, 2, 3].dropFirst(1).dropFirst(1)
```
Comment (moiseev) [inline test/Sema/fixed_ambiguities/rdar35625473.swift]: This is my all times favorite! 🎉

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
Nice!

**Comment by xedin** [comment]
There are some more bugs I've closed but didn't add because they are different variations of the same problem which we have captured here.

**Comment by xedin** [comment]
@swift-ci please test and merge

**Comment by xedin** [comment]
@swift-ci please test and merge

**Pull request merged**